gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-scim/src/main/java/io/gravitee/am/gateway/handler/scim/resources/users/UserEndpoint.java

```diff
@@ -103,39 +102,27 @@ public void get(RoutingContext context) {
      */
     public void update(RoutingContext context) {
         try {
+
             final String body = context.body().asString();
             if (body == null) {
                 context.fail(new InvalidSyntaxException("Unable to parse body message"));
                 return;
             }
+
             final Map<String, Object> payload = Json.decodeValue(body, Map.class);
-            final List<String> schemas = (List<String>) Optional.ofNullable(payload.get("schemas")).orElse(Collections.emptyList());
-            final User user = evaluateUser(schemas, body);
-            final String userId = context.request().getParam("id");
 
-            // username is required
-            if (user.getUserName() == null || user.getUserName().isBlank()) {
-                context.fail(new InvalidValueException("Field [userName] is required"));
-                return;
-            }
+            final SimpleAuthenticationContext authenticationContext = new SimpleAuthenticationContext(new VertxHttpServerRequest(context.request().getDelegate()));
+            authenticationContext.attributes().putAll(context.data());
 
-            // schemas field is REQUIRED and MUST contain valid values and MUST not contain duplicate values
-            try {
-                checkSchemas(user.getSchemas(), Schema.supportedSchemas());
-            } catch (Exception ex) {
-                context.fail(ex);
-                return;
-            }
-
-            // handle identity provider source
-            final String source = userSource(context);
+            final String userId = context.request().getParam("id");
 
             final JWT accessToken = context.get(ConstantKeys.TOKEN_CONTEXT_KEY);
             final String baseUrl = location(context.request());
             principal(accessToken)
                     .map(Optional::ofNullable)
-                    .switchIfEmpty(Maybe.just(Optional.empty()))
-                    .flatMapSingle(optPrincipal -> userService.update(userId, user, source, baseUrl, optPrincipal.orElse(null), context.get(CLIENT_CONTEXT_KEY)))
+                    .switchIfEmpty(Single.just(Optional.empty()))
```

Comment:
Single.Optional.empty sounds weird :p
maybe we can use Maybe?

Reply:
if we are not converting the "empty" maybe in Optional, then we have to duplicate the content of the flatMap operator in a switchIfEmpty.

Wrapping the Optional into a Single Or a Maybe has no really effect as the update action is providing a Single at the end